### PR TITLE
reserve 2% in block hosting volume

### LINKED ISSUE #1309: failed to delete block volumes
### Kind of issue

Bug

### Observed behavior
if we have a block hosting volume of size 200 GB, as we are reserving 1% of it, the creation of block volume is size 198GB is successful, but am not able to delete the block volume

### Expected/desired behavior
if being able to create block volume, I should be able to delete it

### Details on how to reproduce (minimal and precise)
* create block hosting volume of size 200GB
* try to create block volume of size 198 GB
* delete the block volume

### Information about the environment:

- Heketi version used (e.g. v6.0.0 or master):  master
- Operating system used: 
- Heketi compiled from sources, as a package (rpm/deb), or container: 
- If container, which container image: 
- Using kubernetes, openshift, or direct install: 
- If kubernetes/openshift, is gluster running inside kubernetes/openshift or outside: 
- If kubernetes/openshift, how was it deployed (gk-deploy, openshift-ansible, other, custom): 

### Other useful information

below is the report based on my analysis

* block hosting volume of 200GB

am able to create and delete block volumes of size 197 GB.
when I tried to create block volume of size 198 GB, creation will get succeed but however, the deletion will fail from gluster-block side saying No space left on device.

* block hosting volume of 500GB
 
am able to create and delete block volumes of size 490 GB
when I tried to create block volume of size 492 it succeeds but failed to get deleted

### PR DESCRIPTION
**Pull request by Madhu-1**
### What does this PR achieve? Why do we need it?
if we reserve 1% in block hosting volume its not enough for the larger block volumes, at least we should reserve 2% of the block hosting volume we will be able to create a larger block volume.
### Does this PR fix issues?

Fixes #1309

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
@centos-ci ok to tedt

**Comment by raghavendra-talur** [comment]
@centos-ci ok to test

**Comment by nixpanic** [comment]
@centos-ci add to whitelist

**Comment by obnoxxx** [comment]
ssh problem in centos-ci

retest this please

**Comment by Madhu-1** [comment]
@raghavendra-talur we can't print `ReduceRawSize` as the consumable size if block hosting volume already contains block volumes. it's not a good idea to print `ReduceRawSize` as consumable size, we can print free size as the consumable size, whats your thought on this?

**Comment by Madhu-1** [comment]
@obnoxxx am not able to see inline comments for requested changes

File: apps/glusterfs/app_block_volume_test.go [hunk 1]
```diff
@@ -203,7 +203,7 @@ func TestBlockVolumeLargerThanBlockHostingVolume(t *testing.T) {
 	// blockhosting volume and any new blockhosting volume that can
 	// be created
 	request = []byte(`{
-        "size" : 1600
+        "size" : 1079
```
Comment (obnoxxx) [inline apps/glusterfs/app_block_volume_test.go]: Why this change in size?

Comment (Madhu-1) [reply apps/glusterfs/app_block_volume_test.go]: I want to test the corner case, to utilize the completely available free size

File: apps/glusterfs/operations_test.go [hunk 2]
```diff
@@ -1066,7 +1066,7 @@ func TestBlockVolumeCreateOperationTooLargeSizeRequested(t *testing.T) {
 	})
 
 	e := vc.Build()
-	error_string := "The size configured for automatic creation of block hosting volumes (1100) is too small to host the requested block volume of size 1100. Please create a sufficiently large block hosting volume manually."
+	error_string := "The size configured for automatic creation of block hosting volumes (1100) is too small to host the requested block volume of size 1100. The available size on this block hosting volume is 1078. Please create a sufficiently large block hosting volume manually."
```
Comment (raghavendra-talur) [inline apps/glusterfs/operations_test.go]: We need to explicitly tell about the overhead.

File: apps/glusterfs/block_volume_entry.go [hunk 3]
```diff
@@ -376,7 +376,8 @@ func (v *BlockVolumeEntry) Destroy(db wdb.DB, executor executors.Executor) error
 // database operation fails.
 func canHostBlockVolume(tx *bolt.Tx, bv *BlockVolumeEntry, vol *VolumeEntry) (bool, error) {
 	if vol.Info.BlockInfo.FreeSize < bv.Info.Size {
-		logger.Warning("Free size is less than the block volume requested")
+		logger.Warning("Free size %d is less than the requested block volume size %d",
```
Comment (raghavendra-talur) [inline apps/glusterfs/block_volume_entry.go]: lets keep it %v.

Comment (obnoxxx) [inline apps/glusterfs/block_volume_entry.go]: This hunk updates a log message. So this should go into a separate commit with commit message something like "apps:... impove a log message" ... and update the unit tests that cover this message in the same commit. But nothing more.

File: apps/glusterfs/app_block_volume_test.go (hunk 1 above)
Comment (obnoxxx) [reply apps/glusterfs/app_block_volume_test.go]: ah, ok, that's the block volume request size not the BHV - I read this wrong :-)
Makes sense.

**Comment by Madhu-1** [comment]
@obnoxxx added 3 separate commits as requested.
@raghavendra-talur addressed review commands

File: apps/glusterfs/app_block_volume_test.go [hunk 4]
```diff
@@ -213,7 +213,32 @@ func TestBlockVolumeLargerThanBlockHostingVolume(t *testing.T) {
 	body, err := ioutil.ReadAll(io.LimitReader(r.Body, r.ContentLength))
 	tests.Assert(t, err == nil)
 	r.Body.Close()
-	tests.Assert(t, strings.Contains(string(body), "Failed to allocate new block volume: The size configured for automatic creation of block hosting volumes (1100) is too small to host the requested block volume of size 1600. Please create a sufficiently large block hosting volume manually."), "got", string(body))
+	tests.Assert(t, strings.Contains(string(body), "Failed to allocate new block volume: The size configured for automatic creation of block hosting volumes (1100) is too small to host the requested block volume of size 1079. The available size on this block hosting volume, minus overhead, is 1078. Please create a sufficiently large block hosting volume manually."), "got", string(body))
+
+	//check are we able to create a block volume size except reserved 2%
+	request = []byte(`{
+        "size" : 1077
```
Comment (obnoxxx) [inline apps/glusterfs/app_block_volume_test.go]: why test 1077 and not 1078 here? 1078 would exactly be 98% of 1100.

Comment (Madhu-1) [reply apps/glusterfs/app_block_volume_test.go]: already 1GB is created, so 1077
